pydm/widgets/byte.py

```diff
@@ -555,3 +555,422 @@ def alarm_severity_changed(self, new_alarm_severity: int) -> None:
+    Each state represents a different color that can be configured.
+
+    Parameters
+    ----------
+    parent : QWidget
+        The parent widget for the Label
+    init_channel : str, optional
+        The channel to be used by the widget.
+    """
+
+    def __init__(self, parent: Optional[QWidget] = None, init_channel=None):
+        QWidget.__init__(self, parent)
+        PyDMWidget.__init__(self, init_channel=init_channel)
+        self._state_colors = [QColor(Qt.black)] * 16
+        self._curr_state = 0
+        self._curr_color = QColor(Qt.black)
+        self._painter = QPainter()
+        self._brush = QBrush(Qt.SolidPattern)
+        self._pen = QPen(Qt.SolidLine)
+        self._render_as_rectangle = False
+
+    # whether or not we render the widget as a circle (default) or a rectangle
+    @Property(bool)
+    def renderAsRectangle(self) -> bool:
+        return self._render_as_rectangle
+
+    @renderAsRectangle.setter
+    def renderAsRectangle(self, new_val: bool) -> None:
+        if new_val != self._render_as_rectangle:
+            self._render_as_rectangle = new_val
+
+    def value_changed(self, new_val: int) -> None:
+        """
+        Callback invoked when the Channel value is changed.
+        Parameters
+        ----------
+        new_val : int
+            The new value from the channel.
+        """
+        super(PyDMMultiStateIndicator, self).value_changed(new_val)
+        try:
+            int(new_val)
+            if new_val >= 0 and new_val <= 15:  # use states 0-15 (16 total states)
+                self._curr_state = int(new_val)
+                self._curr_color = self._state_colors[self._curr_state]
+                self.update()
+        except Exception:
+            pass
+
+    def paintEvent(self, event: QPaintEvent) -> None:
+        """
+        Paint events are sent to widgets that need to update themselves,
+        for instance when part of a widget is exposed because a covering
+        widget was moved.
+
+        Paint the widget according to the state colors.
+
+        Parameters
+        ----------
+        event : QPaintEvent
+        """
+        self._painter.begin(self)
+        opt = QStyleOption()
+        opt.initFrom(self)
+        self.style().drawPrimitive(QStyle.PE_Widget, opt, self._painter, self)
+        self._painter.setRenderHint(QPainter.Antialiasing)
+        self._painter.setBrush(QBrush(self._curr_color))
+        self._painter.setPen(QPen(Qt.red))
+        if self._render_as_rectangle:
+            self._painter.drawRect(self.rect())
+        else:  # render as circle
+            rect = self.rect()
+            w = rect.width()
+            h = rect.height()
+            r = int(min(w, h) / 2.0 - 2.0 * max(self._pen.widthF(), 1.0))
+            self._painter.drawEllipse(QPoint(w // 2, h // 2), r, r)
+        self._painter.end()
+
+    # color state setters/getters
+    @Property(QColor)
+    def state0Color(self) -> QColor:
```

Comment:
A random thought, for your consideration...

Seeing this sort of stuff makes me wonder if having a `channelValue` Qt property that mirrors the current value of the channel might be something to explore.

I mention that because you can use Qt stylesheets / qss to have property selectors and change any bit of styling as you please, rather than just being limited to color (even if that is technically sufficient for what's being asked here).
So you could imagine:
```
PyDMMultiStateIndicator[currentValue="1"] #something {
    qproperty-penColor: black;
    qproperty-setsomething: foobar;
}
```
We use this method in our vacuum screen widgets, by the way ([ref](https://github.com/pcdshub/vacuumscreens/blob/4dab8dc85e63a13cdb0d01c3dab8dde3cf304e89/styleSheet/masterStyleSheet.qss#L23-L26))